### ISSUE TITLE
Remove challenges when leader is deleted? Or change leader?

### ISSUE THREAD

**Issue by paglias** [HabitRPG/habitica#5979]

**Comment by crookedneighbor** [comment]
I think the leader of the challenge should default to the group leader when the original leader leaves.

**Comment by lemoness** [comment]
I'm not so sure. What if the Challenge is about something that the person
does not like? (For example, if a Challenge about, say, cooking meat was
automatically assigned to a group leader who is a vegetarian, it might make
that user uncomfortable.) This would be even more important for Tavern
Challenges, where presumably one of the admins would be that group leader.

We could make some sort of process where the leader has to approve
ownership, but that might be too complicated. I'd be in favor of deleting
if that's simplest.

To be a little more transparent, maybe the users who are in a Challenge
could be informed if it has been deleted, and then asked if they'd like to
keep or remove the tasks -- same process that occurs when a Challenge ends.
That would also help with the constant questions about how to delete a task
from a deleted Challenge.

**Comment by Alys** [comment]
Deleting challenges when the owner's account is deleted troubles me because we could lose some high quality challenges that way. However Lemoness makes a good point that we can't automatically assign an owner in case the owner doesn't approve of the challenge.

I'd suggest we leave orphaned challenges undeleted and insert the text "no owner" (or similar) where the owner's name usually is -- see the orange arrow:
![screen shot 2015-10-21 at 6 02 46 am](https://cloud.githubusercontent.com/assets/1495809/10619901/85a6c7c4-77bb-11e5-9987-9e8eb4f899af.png)

The Pirates have an offshoot group, the [Elven Grove](https://habitica.com/#/options/groups/guilds/a7d0a7c8-f1cf-4813-b841-75308003f611), dedicated to identifying abandoned challenges and closing them or finding new owners for them. They are happy to take on orphaned challenges, and setting the owner to "no owner" would make it easy for them to find those challenges. Cantras and I are in that guild to assist with actions that can be done by moderators only. You can see a discussion about this in the guild - search for "Issue #5979" to find the first post.

Basically, it means leaving the current situation exactly as it is except that "no owner" is inserted when the owner's name can't be found. This code would need to be modified:
https://github.com/HabitRPG/habitrpg/blob/develop/website/views/options/social/challenges.jade#L191

As an additional enhancement, a system message could be added to the Elven Grove whenever a user who owned challenges deleted their account, but that could be done as a low priority, or not at all.

**Comment by crookedneighbor** [comment]
I like that idea.

**Comment by crookedneighbor** [comment]
It could probably be done like:

``` js
{{challenge.leader.profile.name || env.t('noChallengeOwner')}}
```

**Comment by lemoness** [comment]
That sounds good to me!

**Comment by nicolevonvon** [comment]
I've been directed here from #2044, which seems to be awaiting further feedback. Can I pick it up?

**Comment by crookedneighbor** [comment]
Feel free to pick it up.

**Comment by nicolevonvon** [comment]
Changes made and pull request created.
